src/commands/add.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{
+    manifest::{Manifest, Realm},
+    package_req::PackageReq,
+    package_source::{PackageSource, PackageSourceMap, Registry, TestRegistry},
+    GlobalOptions,
+};
+use anyhow::Context;
+use fs_err as fs;
+use semver::VersionReq;
+use std::{cmp::Ordering, path::PathBuf};
+use structopt::StructOpt;
+
+use super::utils::PackageSpec;
+
+#[derive(Debug, StructOpt)]
+pub struct AddSubcommand {
+    /// Path to the project to add dependencies.
+    #[structopt(long = "project-path", default_value = ".")]
+    pub project_path: PathBuf,
+
+    /// What realm (dev, server, shared) you wish to add your dependency.
+    #[structopt(long = "what-realm", default_value = "shared")]
+    pub what_realm: Realm,
```

Comment:
This seems quite verbose. Is there a reason `--realm` is an insufficient flag?

Reply:
Fixed!